src/project-manager.ts

```diff
@@ -672,18 +668,14 @@ export class InMemoryLanguageServiceHost implements ts.LanguageServiceHost {
 	}
 
 	/**
-	 * @param fileName relative or absolute file path
+	 * @param filePath absolute file path
 	 */
-	getScriptSnapshot(fileName: string): ts.IScriptSnapshot | undefined {
-		let exists = this.fs.fileExists(fileName);
-		if (!exists) {
-			fileName = path.posix.join(this.rootPath, fileName);
```

Comment:
I think this case was reliably absolute paths.
On the other hand, when TS is resolving references a lot more strange paths are requested from the filesystem. I didn't look at TS's logic to understand what it was trying to do, but if we find regressions I would love to document what is happening.